DKDBManager/NSManagedObject+DKDBManager.h

```diff
@@ -18,32 +38,308 @@ typedef NS_ENUM(NSInteger, DKDBManagedObjectState) {
+ *
+ * If the log is activated for the current class, the terminal will display
+ * all important modification.
+ *
+ *
+ * The super function should be called.
+ *
+ * For example, if a `book` does not have a `title` anymore your application
+ * could not want to display/handle the case. Same if its `pages` are missing.
+ *
+ * @code
+ * if let invalidReason = super.invalidReason() {
+ *   return invalidReason
+ * }
+ * if self.pages.count == 0 {
+ *   return "invalid number of pages to display"
+ * }
+ * if self.title == nil {
+ *   return "invalid book title"
+ * }
+ * return nil
+ * @endcode
+ *
+ * @return An invalid reason; nil if valid.
+ */
 - (NSString *)invalidReason;
-- (BOOL)shouldUpdateEntityWithDictionary:(NSDictionary *)dictionary;
-+ (BOOL)verbose;
-+ (NSString *)sortingAttributeName;
-+ (NSPredicate *)primaryPredicateWithDictionary:(NSDictionary *)dictionary;
+
+/**
+ * @brief Override this function to inform the manager whether it should update the current entity or not.
+ *
+ * @param dictionary A NSDictionary object containing information about the database entity to be updated with.
+ *
+ * @discussion Depending on your app and architecture some entities should not be updated or do not need to.
+ * This function allows you to check that and avoid the update of the current entity during the CRUD process.
+ *
+ * For example you might want to verify the `lastUpdate` attribute or a version number from the given dictionary against the current entity.
+ * If they match return then `false` to NOT update the entity and avoid some useless actions.
+ *
+ * The result of this function is ignored when the debug states of the manager are set as
+ * `needForcedUpdate == true` or if `allowUpdate == false`.
+ *
+ * @return TRUE if the entity should be updated by the manager; default is TRUE.
+ */
+ - (BOOL)shouldUpdateEntityWithDictionary:(NSDictionary *)dictionary;
+
+/**
+ * @brief Override this function to enable all activitiy logs about the current class entity or not.
+ *
+ * @return TRUE if log activated; default is FALSE.
+ */
+ + (BOOL)verbose;
+
+/**
+ * @brief Override this function to set a default sorting key for the current class entity.
+ *
+ * @return Sorting key as a NSString object; default is nil.
+ */
+ + (NSString *)sortingAttributeName;
+
+/**
+ * @brief Override this function to create a predicate that will find the right entity corresponding to the given dictionary.
+ *
+ * @param dictionary A NSDictionary object containing information about the database entity to fetched with.
+ *
+ * @discussion This function is called during the CRUD process to fetch an entity from the database.
+ * Depending on the returned predicate the manager will update an existing entity or will create a new one.
+ *
+ * If returns nil then will only take the first entity created (if any) and update it. By doing so only ONE entity will ever be created.`
+ *
+ * If returns a `false predicate` then a new entity will always be created.
```

Comment:
`If [...] returns a ...` replace [...] with `it` or `the method` or something else